packages/discovery-provider/src/gated_content/helpers.py

```diff
@@ -73,8 +74,33 @@ def has_user_purchased_content(
         .filter(
             USDCPurchase.buyer_user_id == user_id,
             USDCPurchase.content_id == content_id,
-            USDCPurchase.content_type == content_type,
+            USDCPurchase.content_type == "track",
```

Comment:
think there is a bug here where if there is a coincidence where:
- the album id happens to have the same id as a track
- the user has access to the track
- the user does not have access to the album
- the user if trying to access ANOTHER track that they don't have access to, via the album

this would return true, where it should be false

Reply:
i think to fix, we should have clear paths for when type is "track" vs "album". should make logic simpler to follow too

Reply:
discussed on huddle

We're only calling this function with a track as input, so there's no risk of this collision.

TODO: add comment to describe the album vs track behavior